cmds/field/field.go

```diff
@@ -0,0 +1,378 @@
+	return lex.typ
+}
+
+func (lex *lexer) peektok() string {
+	return lex.tok
+}
+
+func (lex *lexer) next() (string, toktype) {
+	tok, typ := lex.peek()
+	lex.tok, lex.typ = lex.scan()
+	return tok, typ
+}
+
+func (lex *lexer) scan() (string, toktype) {
+	switch r := lex.nextrune(); {
+	case r == eof:
+		return "", TokEOF
+	case r == ',':
+		return lex.token(), TokComma
+	case r == '|':
+		return lex.token(), TokPipe
+	case r == '-':
+		return lex.token(), TokDash
+	case r == 'N':
+		lex.consume()
+		r = lex.nextrune()
+		if r == 'F' {
+			lex.consume()
+		}
+		lex.ignore()
+		return lex.token(), TokNF
+	case unicode.IsDigit(r):
+		for r := lex.nextrune(); unicode.IsDigit(r); r = lex.nextrune() {
+			lex.consume()
+		}
+		lex.ignore()
+		return lex.token(), TokNum
+	case unicode.IsSpace(r):
+		for r := lex.nextrune(); unicode.IsSpace(r); r = lex.nextrune() {
+			lex.consume()
+		}
+		lex.ignore()
+		return lex.token(), TokSpace
+	default:
+		fatal("Lexical error at character '%v'", r)
+	}
+	return "", TokError
+}
+
+func (lex *lexer) nextrune() (r rune) {
+	if lex.pos >= len(lex.input) {
+		lex.width = 0
+		return eof
+	}
+	r, lex.width = utf8.DecodeRuneInString(lex.input[lex.pos:])
+	return r
+}
+
+func (lex *lexer) consume() {
+	lex.pos += lex.width
+	lex.width = 0
+}
+
+func (lex *lexer) ignore() {
+	lex.width = 0
+}
+
+func (lex *lexer) token() string {
+	lex.consume()
+	tok := lex.input[lex.start:lex.pos]
+	lex.start = lex.pos
+	return tok
+}
+
+func process(file *os.File, rv []frange, delim *regexp.Regexp, outsep string, collapse bool) {
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		prefix := ""
+		printed := false
+		line := scanner.Text()
+		fields := split(line, delim, collapse)
+		for _, r := range rv {
+			begin, end := r.begin, r.end
+			switch {
+			case begin == 0:
+				pprefix(prefix)
+				prefix = outsep
+				fmt.Print(line)
+				printed = true
+				begin = 1
+			case begin == LastField:
+				begin = len(fields) -1
+			case begin < 0:
+				begin += len(fields)
+			default:
+				begin--
+			}
+			if end < 0 {
+				end += len(fields) + 1
+			}
+			if begin < 0 || end < 0 || end < begin || len(fields) < begin {
+				continue
+			}
+			for i := begin; i < end && i < len(fields); i++ {
+				pprefix(prefix)
+				prefix = outsep
+				fmt.Print(fields[i])
+				printed = true
+			}
+		}
+		if printed || !collapse {
+			fmt.Println()
+		}
+	}
+	err := scanner.Err()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+	}
+}
+
+func split(s string, delim *regexp.Regexp, collapse bool) []string {
+	sv := delim.Split(s, -1)
+	if !collapse {
+		return sv
+	}
+	rv := []string{}
+	for _, s := range sv {
+		if s != "" {
+			rv = append(rv, s)
+		}
+	}
+	return rv
+}
+
+func pprefix(prefix string) {
+	if prefix == "" {
+		return
+	}
+	if *nuloutsepflag {
+		fmt.Print("\x00")
+	} else {
+		fmt.Print(prefix)
+	}
+}
+
+func fatal(format string, a ...interface{}) {
+	fmt.Fprintf(os.Stderr, format+"\n", a...)
+	usage()
+}
+
+func usage() {
```

Comment:
I realize it looks a bit nasty, but see how to set up a usage in rm/fm.go::init. 

The reason to do it this way is to make it so we can use flag.Usage() as well as a custom usage function, and it will all work in busybox mode.